generator/main.go

```diff
@@ -0,0 +1,485 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.universe.tf/metallb/api/v1beta1"
+	"go.universe.tf/metallb/api/v1beta2"
+	"go.universe.tf/metallb/internal/config"
+	"go.universe.tf/metallb/internal/pointer"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	resourcesNameSpace = "metallb-system"
+	spacer             = "---\n"
+	autoGenComment     = "# This was autogenerated by MetalLB's custom resource generator.\n"
+)
+
+var (
+	inputDirPath = "/var/input"
+)
+
+func main() {
+	var origin = flag.String("origin", "./config.yaml", "name of the configmap file to covnert")
+	flag.Parse()
+
+	resourcesYAML, err := generate(*origin)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Println(resourcesYAML)
+}
+
+// generate gets a name of a metallb configmap file, converts it to
+// the matching metallb custom resources yamls, and returns it as a string.
+func generate(origin string) (string, error) {
```

Comment:
You can pass a Writer to generate, so you can choose from outside whether write to a file or to stdoutput.

Reply:
how would we let the end-user choose?
maybe the default is stdout and optional adding --out=filename to write to a file?

Reply:
I would not, but the writer comes handy in test scenarios.